stellar-contract-env-common/src/unimplemented_env.rs

```diff
@@ -10,6 +10,10 @@ impl EnvBase for UnimplementedEnv {
     }
 
     fn check_same_env(&self, _other: &Self) {}
+
+    fn deep_clone(&self) -> Self {
+        Self
```

Comment:
Given that it would be a logical error to call this on a Guest, is there a way for us to make it a compile error to that ever to happen?

Reply:
This comment was intended to be on the Guest impl.